src/java/org/apache/cassandra/db/guardrails/GuardrailsConfig.java

```diff
@@ -429,4 +429,40 @@ void setMaximumTimestampThreshold(@Nullable DurationSpec.LongMicrosecondsBound w
      */
     void setMinimumTimestampThreshold(@Nullable DurationSpec.LongMicrosecondsBound warn,
                                       @Nullable DurationSpec.LongMicrosecondsBound fail);
+
+    /**
+     * @return The threshold to warn when number of referenced indexes
+     * to be searched for non-partition restricted query is greater than treshold.
+     */
+    int getSaiSSTableIndexesPerQueryWarnThreshold();
+
+    /**
+     * @return The threshold to fail when number of referenced indexes
+     * to be searched for non-partition restricted query is greater than treshold.
+     */
+    int getSaiSSTableIndexesPerQueryFailThreshold();
+
+    /**
+     * Sets warn and fail thresholds for a guardrail checking number of referenced indexes
+     * to search for non-partition restricted query.
+     * @param warn value to set for warn threshold
+     * @param fail value to set for fail threshold
+     */
```

Comment:
Suggestion:
```
/**
  * Sets warning and failure thresholds for the number of SAI SSTable indexes searched by a query
  * ...
  */
```

Reply:
For all three of these, big thing is removing the language around partition restrictions, which don't come into play explicitly for this guardrail.